### PR TITLE
Update libpq version 11.7 and 12.2 + Package libpgport

### PR DESCRIPTION
**Pull request by syoliver**
Specify library name and version:  **libpq/12.2**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libpq/all/conanfile.py [hunk 1]
```diff
@@ -113,12 +113,16 @@ def build(self):
...
+                        self.run("perl build.pl libpgport")
         else:
             autotools = self._configure_autotools()
             with tools.chdir(os.path.join(self._source_subfolder, "src", "backend")):
                 autotools.make(target="generated-headers")
             with tools.chdir(os.path.join(self._source_subfolder, "src", "common")):
                 autotools.make()
+            if self.version >= '12':
+              with tools.chdir(os.path.join(self._source_subfolder, "src", "port")):
+                  autotools.make()
```
Comment (uilianries) [inline recipes/libpq/all/conanfile.py:125]:
```suggestion
            if tools.Version(self.version) >= '12':
                with tools.chdir(os.path.join(self._source_subfolder, "src", "port")):
                    autotools.make()
```

**Comment by madebr** [comment]
@syoliver 
What do you think about my comment in the other libpq pr?
https://github.com/conan-io/conan-center-index/pull/1620#issuecomment-627412925

**Comment by syoliver** [comment]
> @syoliver 
> What do you think about my comment in the other libpq pr?
I don't know exactly what is the problem with libpgport. I needed it in my first attempt to use libpq on windows in my soci PR and it seems that I don't need it anymore with your rewrite of soci recipe. I have to check if my hacked libpq package was not in cache yesterday. 
Basically, I will need to check deeper if I really need to merge this PR with the one you linked and drop this one. If libpgport is needed, you may have to add it in windows build tool.

**Comment by madebr** [comment]
Symbols of the static port library might not be used by the test_package, so no linking errors appear.
But a bigger project might run into problems.
The fact that the shared libpq library does not build without the port library is a clear indication that some function depends on the port library.

**Comment by syoliver** [comment]
> Symbols of the static port library might not be used by the test_package, so no linking errors appear.
> But a bigger project might run into problems.
> The fact that the shared libpq library does not build without the port library is a clear indication that some function depends on the port library.

I was thinking that is may also depend of actual system library of the packaging host compiler versus package host user... If we may force compilation of compatibility layer we would be safer. But I am not sure if it make sense or if it is actually a good idea.

**Pull request closed without merging**